packages/dd-trace/src/require-package-json.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+const path = require('path')
+const fs = require('fs')
+
+/**
+ * Given a package name and a module to start from, find a package's
+ * package.json file, parses it, and returns the result.
+ *
+ * Equivalent to require(`${name}/package.json`) prior to Node 12.
```

Comment:
Since we already support newer versions of Node, why is this change needed for this PR? Is it a bug fix for something that should already be working and isn't? If that's the case, should it be its own PR? It's not a large change but since it touches the loader I'd move it out if it's not related to dropping Node 8/10.

Reply:
>  Is it a bug fix for something that should already be working and isn't? If that's the case, should it be its own PR?

Sort of. Because we generally test things on our minimum support versions, we weren't testing on Node 12 for a lot of things, which introduces the new behaviour with the `"exports"` field. This means that yes, we had a bug here that was in our testing blindspot. By moving to Node 12, it's no longer in our testing blind spot, and we need a fix for CI to pass at all, so here we are.

I could do this in a separate PR but:

1. We'd have to test on Node 12 in that PR. Without having a full testing matrix, that's a bit difficult right now.
2. That PR would have to land before this one.